api/src/paths/projects/{projectId}/artifacts/list.ts

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { WRITE_ROLES } from '../../../../constants/roles';
+import { getFileListFromS3 } from '../../../../utils/file-utils';
+import { getLogger } from '../../../../utils/logger';
+
+const defaultLog = getLogger('/api/projects/{projectId}/artifacts/list');
+
+export const GET: Operation = [getMediaList()];
+
+GET.apiDoc = {
+  description: 'Fetches a list of artifact descriptions in a project.',
+  tags: ['artifacts'],
+  security: [
+    {
+      Bearer: WRITE_ROLES
+    }
+  ],
+  parameters: [
+    {
+      in: 'path',
+      name: 'projectId',
+      required: true
+    }
+  ],
+  responses: {
+    200: {
+      description: 'Project get response file description array.',
+      content: {
+        'application/json': {
+          schema: {
+            type: 'array',
+            items: {
+              type: 'object',
+              properties: {
+                fileName: {
+                  description: 'The file name of the artifact',
+                  type: 'string'
+                },
+                lastModified: {
+                  description: 'The date the object was last modified',
+                  type: 'string'
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    503: {
+      $ref: '#/components/responses/503'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+function getMediaList(): RequestHandler {
+  return async (req, res) => {
+    defaultLog.debug('started... req.params.projectId=' + req.params.projectId);
+
+    if (!req.params.projectId) {
+      throw {
+        status: 400,
+        message: 'Missing required path param `projectId`'
+      };
+    }
+
+    const { Contents } = await getFileListFromS3(req.params.projectId + '/');
```

Comment:
Related to this comment: https://github.com/bcgov/biohubbc/pull/61/files#r574888254

If it now returns null, we can then update this to be something like:

```Typescript
const ContentsList = await getFileListFromS3(req.params.projectId + '/');
if(!ContentsList) {
// do error stuff
}
const contents = ContentsList.contents;
```

And then further down, rather than casting the content to any
```Typescript
      Contents.forEach(function (content: any) {
```
we should just be able to leave it, as the code will no that its an S3 Object type (since we already ruled out it being null)